src/tests/stats/test_mean.f90

```diff
@@ -79,4 +90,20 @@ program test_mean
 call assert( sum( abs( mean(d4,3) - sum(d4,3)/real(size(d4,3), dp) )) < dptol)
 call assert( sum( abs( mean(d4,4) - sum(d4,4)/real(size(d4,4), dp) )) < dptol)
 
+! check mask = .false.
+
+call assert( isnan(mean(d4, .false.)))
+call assert( any(isnan(mean(d4, 1, .false.))))
+call assert( any(isnan(mean(d4, 2, .false.))))
+call assert( any(isnan(mean(d4, 3, .false.))))
+call assert( any(isnan(mean(d4, 4, .false.))))
+
+
+! check mask of the same shape as input
+call assert( abs(mean(d4, d4 > 0._dp) - sum(d4, d4 > 0._dp)/real(count(d4 > 0._dp), dp)) < dptol)
```

Comment:
Your code is correct, but one can simplify it as:
```suggestion
call assert( abs(mean(d4, d4 > 0) - sum(d4, d4 > 0)/real(count(d4 > 0), dp)) < dptol)
```
So I usually like it that way better. But that's up to you.

Reply:
modified!